src/inline_module.cpp

```diff
@@ -0,0 +1,158 @@
+        // a literal
+        if(arg_cond.empty())
+        {
+            std::vector<argument> arg_outs;
+            for(const auto& mod : mod_inputs)
+            {
+                auto last = std::prev(mod->end());
+                std::vector<instruction_ref> mod_outputs;
+                if(last->name() == "@return")
+                {
+                    mod_outputs = last->inputs();
+                }
+                else
+                {
+                    // no return instruction, last instruction is output
+                    mod_outputs.push_back(last);
+                }
+
+                // only one output is considered for now
+                for(const auto& out : mod_outputs)
+                {
+                    auto mod_out = out->eval();
+                    if(mod_out.empty())
+                        return;
+                    arg_outs.push_back(mod_out);
+                }
+            }
+
+            auto out_num            = arg_outs.size() / 2;
+            auto cond               = ins->inputs().front();
+            const auto& ins_outputs = ins->outputs();
+            auto icond              = m.insert_instruction(
+                ins, make_op("convert", {{"target_type", shape::float_type}}), cond);
+            for(std::size_t i = 0; i < out_num; ++i)
+            {
+                auto l0 = m.add_literal(literal(arg_outs.at(i).get_shape(), arg_outs.at(i).data()));
+                auto l1 = m.add_literal(
+                    literal(arg_outs.at(i + out_num).get_shape(), arg_outs.at(i + out_num).data()));
+                auto lens  = l0->get_shape().lens();
+                auto mcond = m.insert_instruction(
+                    ins, make_op("multibroadcast", {{"output_lens", lens}}), icond);
+                std::size_t elem_num = l0->get_shape().elements();
+                std::vector<float> vec_offset(elem_num, elem_num);
+                std::vector<float> vec_ind(elem_num);
+                shape sidx{shape::float_type, lens};
+                std::iota(vec_ind.begin(), vec_ind.end(), elem_num);
+                auto lidx    = m.add_literal(literal(sidx, vec_ind));
+                auto loffset = m.add_literal(literal(sidx, vec_offset));
+                auto moffset = m.insert_instruction(ins, make_op("mul"), mcond, loffset);
+                auto f_ind   = m.insert_instruction(ins, make_op("sub"), lidx, moffset);
+                auto ins_ind = m.insert_instruction(
+                    ins, make_op("convert", {{"target_type", shape::int32_type}}), f_ind);
+                auto l01 = m.insert_instruction(ins, make_op("concat", {{"axis", 0}}), l0, l1);
+                auto rl  = m.insert_instruction(
+                    ins, make_op("reshape", {{"dims", {l0->get_shape().elements() * 2}}}), l01);
+                auto r = m.insert_instruction(ins, make_op("gather", {{"axis", 0}}), rl, ins_ind);
+                m.replace_instruction(ins_outputs.at(i), r);
```

Comment:
Why is all this needed? Shouldn't it just insert a `make_tuple` of the multiple arguments?

Reply:
The reason is that the gather does not work for the tuple type.